src/components/icon/js/iconDirective.js

```diff
@@ -230,25 +230,12 @@ function mdIconDirective($mdIcon, $mdTheming, $mdAria ) {
       return false;
     }
 
-    function prepareForFontIcon () {
+    function prepareForFontIcon() {
       if (!scope.svgIcon && !scope.svgSrc) {
-
         if (scope.fontIcon) {
           element.addClass('md-font ' + scope.fontIcon);
         }
-
-        if (scope.fontSet) {
-          element.addClass($mdIcon.fontSet(scope.fontSet));
-        }
-
-        if (shouldUseDefaultFontSet()) {
```

Comment:
The `!scope.fontSet` component of the boolean expression in `shouldUseDefaultFontSet` blocked the default fontSet from being applied when it should have been (the original bug).

The remaining `!scope.fontIcon` component proved to be unnecessary because when `scope.fontSet` is undefined, `$mdIcon.fontSet(scope.fontSet)` is equivalent to `$mdIcon.fontSet()`. This allowed me to remove `shouldUseDefaultFontSet` and combine lines 240-247 into one expression.